src/types/gecko-profile.js

```diff
@@ -0,0 +1,140 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import type { Lib, IndexIntoStringTable } from './profile';
+import type { MarkerPayload } from './markers';
+import type { Milliseconds } from './units';
+
+export type IndexIntoGeckoFrameTable = number;
+export type IndexIntoGeckoStackTable = number;
+
+export type GeckoMarkers = {
+  schema: { name: 0, time: 1, data: 2 },
+  data: Array<[IndexIntoStringTable, Milliseconds, MarkerPayload]>,
+};
+
+/**
+ * These structs aren't very DRY, but it is a simple and complete approach.
+ * These structs are the initial transformation of the Gecko data to the
+ * processed format.
+ */
+export type GeckoMarkerStruct = {
+  name: IndexIntoStringTable[],
+  time: Milliseconds[],
+  data: MarkerPayload[],
+  length: number,
+};
+
+export type GeckoSamples = {
+  schema: {
+    stack: 0,
+    time: 1,
+    responsiveness: 2,
+    rss: 3,
+    uss: 4,
+  },
+  data: Array<
+    [
+      null | IndexIntoGeckoStackTable,
+      Milliseconds, // since profile.meta.startTime
+      // milliseconds since the last event was processed in this
+      // thread's event loop at the time that the sample was taken
+      Milliseconds,
+      any, // TODO
+      any, // TODO
```

Comment:
I'd like this as a follow-up, along with docs explaining what it is since I'm just copying what we have in the processed `Profile` type definition, and we aren't currently using this information.